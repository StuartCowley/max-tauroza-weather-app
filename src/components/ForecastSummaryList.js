import React from "react";
import ForecastSummary from "./ForecastSummary";

const ForecastSummaryList = ({ forecasts }) => {
  return (
    <div className="forecast-summary-list">
      {forecasts.map((forecast) => (
        <ForecastSummary key={forecast.date} forecast={forecast} />
      ))}
    </div>
  );
};

export default ForecastSummaryList;
