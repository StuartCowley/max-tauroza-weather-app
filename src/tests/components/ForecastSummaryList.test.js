import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaryList from "../../components/ForecastSummaryList";

describe("Forecast Summary List", () => {
  const validProps = [
    {
      date: 1111111,
      description: "Stub description",
      icon: "stubIcon",
      temperature: {
        min: 12,
        max: 22,
      },
    },
    {
      date: 1111112,
      description: "Stub description",
      icon: "stubIcon",
      temperature: {
        min: 12,
        max: 22,
      },
    },
    {
      date: 1111113,
      description: "Stub description",
      icon: "stubIcon",
      temperature: {
        min: 12,
        max: 22,
      },
    },
    {
      date: 1111114,
      description: "Stub description",
      icon: "stubIcon",
      temperature: {
        min: 12,
        max: 22,
      },
    },
    {
      date: 1111115,
      description: "Stub description",
      icon: "stubIcon",
      temperature: {
        min: 12,
        max: 22,
      },
    },
  ];
  it("renders list correctly", () => {
    const { asFragment } = render(
      <ForecastSummaryList forecasts={validProps} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
