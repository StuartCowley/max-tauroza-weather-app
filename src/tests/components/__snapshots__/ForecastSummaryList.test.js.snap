// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Summary List renders list correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary-list"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <img
          alt="Weather icon"
          src="https://cdn-icons-png.flaticon.com/128/365/365851.png"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary_description"
      >
        Stub description
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <img
          alt="Weather icon"
          src="https://cdn-icons-png.flaticon.com/128/365/365925.png"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary_description"
      >
        Stub description
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <img
          alt="Weather icon"
          src="https://cdn-icons-png.flaticon.com/128/365/365867.png"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary_description"
      >
        Stub description
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <img
          alt="Weather icon"
          src="https://cdn-icons-png.flaticon.com/128/365/365925.png"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary_description"
      >
        Stub description
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <img
          alt="Weather icon"
          src="https://cdn-icons-png.flaticon.com/128/365/365859.png"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary_description"
      >
        Stub description
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
